### ISSUE TITLE
GlobalEventHandlers.onerror should be of type ErrorEventHandler | null

### ISSUE THREAD

**Issue by thdk** [Microsoft/TypeScript#29876]
**TypeScript Version:**  3.2.1

**Search Terms: lib.dom.ts GlobalEventHandlers onerror**

**Code**

```ts
const imageEl = document.createElement("img");

// enable strictNullChecks in options and below line will give a type error
// type 'null' is not assignable to type 'ErrorEventHandler'
imageEl.onerror = null;
```

**Expected behavior:**
it should be possible to assign null to onerror

**Actual behavior:**
type 'null' is not assignable to type 'ErrorEventHandler'

**Playground Link:** 
[Playground link](https://typescript-play.js.org/#code/MYewdgzgLgBAlgWwIYHMCmBRANjAvDAExGAFcE0woA6YAJzSSkyzXMoAoAiRFTgSgDcAKCEB6UTApIARixjRacYFAByJLFgDCACzTAA1hHhgYIAA5Q44I0jAEY0tFhAB3GFjhg0MF3A0wUOAA3byQYKABPM280WloQWjEJSOiYAHIwdSw0+CMwEFgkCAg4FDAZOSgQcKjvNIw4hIwQygAJWwIWWjShRFRmKnBY+No8GEyNASA)

**Related Issues:** 
https://github.com/Microsoft/TypeScript/issues/28638 (not really related)

-------

I believe line https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L5478 can be modified to allow null as follows:
**Option 1:**
```
/**
     * Fires when an error occurs during object loading.
     * @param ev The event.
     */
    onerror: ErrorEventHandler | null;
```

**Option 2**
```
interface ErrorEventHandlerNonNull {
    (event: Event | string, source?: string, fileno?: number, columnNumber?: number, error?: Error): void;
}
```
and a type ErrorEventHandler
```
type ErrorEventHandler = ErrorEventHandlerNonNull | null;
```
and leave  'onerror: ErrorEventHandler;' as is.

**Comment by Fabianopb** [comment]
I've some spare time these weeks, I could take a look at this one too!

**Comment by Fabianopb** [comment]
I've checked the issue, apparently this is already implemented in the latest version (`3.3.3333`) using your *Option2*.

@thdk could you please check the following references and confirm that?

https://github.com/Microsoft/TypeScript/blob/v3.3.3333/lib/lib.dom.d.ts#L17115
https://github.com/Microsoft/TypeScript/blob/v3.3.3333/lib/lib.dom.d.ts#L17827

**Comment by thdk** [comment]
@Fabianopb seems to fixed indeed in 3.3.3333 👍 

I'll close the issue.
